### PR TITLE
[Fix] Remove the legacy hack function for monocular dataset browsing

### PR DESCRIPTION
**Pull request by Tai-Wang**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Please describe the motivation of this PR and the goal you want to achieve through this PR.
Remove the legacy hack function for monocular dataset browsing.

## Modification
Please briefly describe what modification is made in this PR.
Only remove an unused function.

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by Wuziyi616 — APPROVED** [review]

**Review by Wuziyi616 — COMMENTED** [review]
Need to resolve conflict and unit test?

**Comment by Wuziyi616** [comment]
There are some other imports like [here](https://github.com/Tai-Wang/mmdetection3d/blob/1ea81878ee663b441087ceafff77de06237faa63/mmdet3d/apis/inference.py#L430), [here](https://github.com/Tai-Wang/mmdetection3d/blob/1ea81878ee663b441087ceafff77de06237faa63/mmdet3d/core/bbox/__init__.py#L14), [here](https://github.com/Tai-Wang/mmdetection3d/blob/1ea81878ee663b441087ceafff77de06237faa63/mmdet3d/datasets/nuscenes_mono_dataset.py#L14) and [here](https://github.com/Tai-Wang/mmdetection3d/blob/1ea81878ee663b441087ceafff77de06237faa63/mmdet3d/models/detectors/single_stage_mono3d.py#L8)

**Review by Wuziyi616 — APPROVED** [review]
I think this PR is ready for merge @ZwwWayne

File: mmdet3d/core/bbox/structures/utils.py [hunk 1]
```diff
@@ -147,46 +147,3 @@ def points_cam2img(points_3d, proj_mat, with_depth=False):
     if with_depth:
         return torch.cat([point_2d_res, point_2d[..., 2:3]], dim=-1)
     return point_2d_res
-
-
-def mono_cam_box2vis(cam_box):
```
Comment (ZwwWayne) [inline mmdet3d/core/bbox/structures/utils.py:152]: This module should be kept and pop deprecated warning for 3 months.

File: mmdet3d/core/bbox/structures/utils.py [hunk 2]
```diff
@@ -167,6 +168,8 @@ def mono_cam_box2vis(cam_box):
     Returns:
         :obj:`CameraInstance3DBoxes`: Box after conversion.
     """
+    warning.warn('DeprecationWarning: The hack for visualizing monocular 3D '
```
Comment (ZwwWayne) [inline mmdet3d/core/bbox/structures/utils.py:171]: The warning is incorrect, we should directly say that this function will be deprecated soon.

**Pull request merged**